nautobot/ipam/models.py

```diff
@@ -612,18 +612,24 @@ def delete(self, *args, **kwargs):
             protected_objects.update(parent=self.parent)
             return super().delete(*args, **kwargs)
 
+    def get_parent(self):
+        # Determine if a parent exists and set it to the closest ancestor by `prefix_length`.
+        if supernets := self.supernets():
+            return max(supernets, key=operator.attrgetter("prefix_length"))
+        return None
+
+    def clean(self):
+        self.parent = self.get_parent()
+        return super().clean()
+
     def save(self, *args, **kwargs):
         if isinstance(self.prefix, netaddr.IPNetwork):
             # Clear host bits from prefix
             # This also has the subtle side effect of calling self._deconstruct_prefix(),
             # which will (re)set the broadcast and ip_version values of this instance to their correct values.
             self.prefix = self.prefix.cidr
 
-        # Determine if a parent exists and set it to the closest ancestor by `prefix_length`.
-        supernets = self.supernets()
-        if supernets:
-            parent = max(supernets, key=operator.attrgetter("prefix_length"))
-            self.parent = parent
+        self.parent = self.get_parent()
```

Comment:
Do we need this logic here in save() after we already have it in clean()?

Reply:
Yes, we still need it, incase user uses `.save` instead of `.validated_save`